node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -984,21 +968,44 @@ impl StandardGossipHandler {
     }
 
     fn compute_patch(
+        &self,
+        agrs: &Vec<AccessibleGossipRecord>,
+        root_node: &NodeRecord,
+    ) -> HashSet<PublicKey> {
+        let agrs_by_key = agrs
+            .iter()
+            .map(|agr| (&agr.inner.public_key, agr))
+            .collect::<HashMap<&PublicKey, &AccessibleGossipRecord>>();
+
+        let mut patch: HashSet<PublicKey> = HashSet::new();
+        self.compute_patch_recursive(
+            &mut patch,
+            root_node.public_key(),
+            &agrs_by_key,
+            DEFAULT_MINIMUM_HOP_COUNT,
+            root_node,
+        );
+
+        patch
+    }
+
+    fn compute_patch_recursive(
         &self,
         patch: &mut HashSet<PublicKey>,
         node: &PublicKey,
         agrs: &HashMap<&PublicKey, &AccessibleGossipRecord>,
         hops_remaining: usize,
-        database: &NeighborhoodDatabase,
+        root_node: &NodeRecord,
     ) {
         patch.insert(node.clone());
-        if hops_remaining > 0 {
-            let neighbors = if patch.len() == 1 {
-                // Root Node
-                database.get_all_half_neighbors()
+        if hops_remaining == 0 {
+            return;
+        } else {
+            let neighbors = if node == root_node.public_key() {
+                root_node.inner.neighbors.clone()
```

Comment:
Maybe these `clone()`s aren't necessary, since we're using `&neighbors` below instead of `neighbors`.